flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/tasks/SourceStreamTask.java

```diff
@@ -45,6 +45,8 @@
 public class SourceStreamTask<OUT, SRC extends SourceFunction<OUT>, OP extends StreamSource<OUT, SRC>>
 	extends StreamTask<OUT, OP> {
 
+	private static final Runnable SOURCE_POISON_LETTER = () -> {};
```

Comment:
Can these special mail instances put in a central place in `MailboxImpl.java` to be more clear?

Reply:
I think that would be to wrong place: this letter is not special to the mailbox, it is only special to this task.